currency_monthly_rate/models/res_currency.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api, _
+
+
+class ResCurrency(models.Model):
+    _inherit = "res.currency"
+
+    monthly_rate = fields.Float(compute='_compute_current_monthly_rate',
+                                string='Current Monthly Rate', digits=(12, 6),
+                                help='The monthly rate of the currency to '
+                                     'the currency of rate 1.')
+    monthly_rate_ids = fields.One2many('res.currency.rate.monthly',
+                                       'currency_id', string='Monthly rates')
+
+    @api.multi
+    def _compute_current_monthly_rate(self):
+        date = self.env.context.get('date') or fields.Date.today()
+        company_id = self.env.context.get('company_id') or self.env[
+            'res.users']._get_company().id
+        # the subquery selects the first rate before 'date' for the given
```

Comment:
@grindtildeath Can we just replace here the sql query with a monthly_rate_ids.filtered(...) since we already have the one2many field.

Reply:
@feketemihai Thanks for your review. The idea was to have the same behaviour as res.currency, so I did keep the same SQL query changing the table from res_currency_rate to res_currency_rate_monthly.